tests/components/icloud/test_config_flow.py

```diff
@@ -273,34 +284,52 @@ def init_config_flow(hass: HomeAssistantType):
     hass: HomeAssistantType, service_send_verification_code_failed: MagicMock
 ):
     """Test when we have errors during send_verification_code."""
-    flow = init_config_flow(hass)
-    await flow.async_step_user({CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
 
-    result = await flow.async_step_trusted_device({CONF_TRUSTED_DEVICE: 0})
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {CONF_TRUSTED_DEVICE: 0}
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == CONF_TRUSTED_DEVICE
     assert result["errors"] == {CONF_TRUSTED_DEVICE: "send_verification_code"}
 
 
-async def test_verification_code(hass: HomeAssistantType):
+async def test_verification_code(hass: HomeAssistantType, service: MagicMock):
     """Test verification_code step."""
-    flow = init_config_flow(hass)
-    await flow.async_step_user({CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {CONF_TRUSTED_DEVICE: 0}
+    )
 
-    result = await flow.async_step_verification_code()
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == CONF_VERIFICATION_CODE
 
 
-async def test_verification_code_success(
-    hass: HomeAssistantType, service_with_cookie: MagicMock
-):
+async def test_verification_code_success(hass: HomeAssistantType, service: MagicMock):
     """Test verification_code step success."""
-    flow = init_config_flow(hass)
-    await flow.async_step_user({CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {CONF_TRUSTED_DEVICE: 0}
+    )
 
-    result = await flow.async_step_verification_code({CONF_VERIFICATION_CODE: 0})
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {CONF_VERIFICATION_CODE: 0}
```

Comment:
The verification code must be a string according to the validation schema.

Reply:
Oh yeah, that make the error log much clearer !

A proof that testing the config entry flow is much better, thanks.